core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -194,7 +241,10 @@ private Translog openTranslog(EngineConfig engineConfig, IndexWriter writer, boo
             }
         }
         final Translog translog = new Translog(translogConfig);
+        final Translog.TranslogGeneration generation = translogConfig.getTranslogGeneration();
         if (generation == null || generation.translogUUID == null) {
+            assert openMode != EngineConfig.OpenMode.OPEN_INDEX_AND_TRANSLOG : "OpenMode must not be "
+                + EngineConfig.OpenMode.OPEN_INDEX_AND_TRANSLOG;
```

Comment:
well if the assertion fails it's `EngineConfig.OpenMode.OPEN_INDEX_AND_TRANSLOG` so it's there already?